datasources-agroal/src/test/java/org/wildfly/extension/datasources/agroal/AgroalTransformersTestCase.java

```diff
@@ -0,0 +1,139 @@
+import org.jboss.dmr.ModelNode;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test transformation between model versions
+ *
+ * @author <a href="lbarreiro@redhat.com">Luis Barreiro</a>
+ */
+public class AgroalTransformersTestCase extends AbstractSubsystemBaseTest {
+
+    public AgroalTransformersTestCase() {
+        super(AgroalExtension.SUBSYSTEM_NAME, new AgroalExtension());
+    }
+
+    private static String getExtensionMavenGAV(ModelTestControllerVersion version) {
+        if (version.isEap()) {
+            return "org.jboss.eap:wildfly-datasources-agroal:" + version.getMavenGavVersion();
+        }
+        return "org.wildfly:wildfly-datasources-agroal:" + version.getMavenGavVersion();
+    }
+
+    @Override
+    protected AdditionalInitialization createAdditionalInitialization() {
+        // Create a AdditionalInitialization.MANAGEMENT variant that has all the external capabilities used by the various configs used in this test class
+        return AdditionalInitialization.withCapabilities(
+                AbstractDataSourceDefinition.AUTHENTICATION_CONTEXT_CAPABILITY + ".secure-context",
+                CredentialReference.CREDENTIAL_STORE_CAPABILITY + ".test-store"
+        );
+    }
+
+    @Override
+    protected String getSubsystemXml() throws IOException {
+        return readResource("agroal_2_0-full.xml");
+    }
+
+    @Override
+    protected String getSubsystemXsdPath() throws IOException {
+        return "schema/wildfly-agroal_2_0.xsd";
+    }
+
+    @Test
+    public void testTransformers_2_0_0() throws Exception {
+        testTransformers(ModelTestControllerVersion.EAP_7_2_0, AgroalExtension.VERSION_1_0_0, "agroal_2_0-transform.xml");
+    }
+
+    private void testTransformers(ModelTestControllerVersion controllerVersion, ModelVersion agroalVersion, String xmlResource) throws Exception {
+        //Boot up empty controllers with the resources needed for the ops coming from the xml to work
+        KernelServicesBuilder builder = createKernelServicesBuilder(createAdditionalInitialization());
+
+        builder.createLegacyKernelServicesBuilder(createAdditionalInitialization(), controllerVersion, agroalVersion)
+               .addMavenResourceURL(getExtensionMavenGAV(controllerVersion))
+               .skipReverseControllerCheck()
+               .dontPersistXml();
+
+        KernelServices mainServices = builder.build();
+        assertTrue(mainServices.isSuccessfulBoot());
+        assertTrue(mainServices.getLegacyServices(agroalVersion).isSuccessfulBoot());
+
+        List<ModelNode> ops = builder.parseXmlResource(xmlResource);
+        System.out.println("ops = " + ops);
+        PathAddress subsystemAddress = PathAddress.pathAddress(SUBSYSTEM, AgroalExtension.SUBSYSTEM_NAME);
```

Comment:
Once you have fixed https://github.com/wildfly/wildfly/pull/11776/files#r264739494 you will need to rename this variable and append the path for the xa-datasource children. I'm afraid I don't remember if this stuff allows wildcards or not. If it doesn't you will need to append the proper resulting path element from our configuration.